src/components/product-filters/product-filters.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { useNavigation } from '@remix-run/react';
+import { IProductFilters } from '~/api/types';
+import { Accordion } from '../accordion/accordion';
+import { RangeSlider } from '../range-slider/range-slider';
+
+interface ProductFiltersProps {
+    appliedFilters: IProductFilters;
+    onFiltersChange: (filters: IProductFilters) => void;
+    lowestPrice: number;
+    highestPrice: number;
+    formatPrice?: (value: number) => string;
+}
+
+export const ProductFilters = ({
+    appliedFilters,
+    onFiltersChange,
+    lowestPrice,
+    highestPrice,
+    formatPrice,
+}: ProductFiltersProps) => {
+    const navigation = useNavigation();
+
+    const [filters, setFilters] = useState(appliedFilters);
```

Comment:
I don't quite understand this. If we're always updating state based on `appliedFilters`, then why do we need the internal state variable? Is it because `appliedFilters` updates with a delay?

Reply:
> Is it because appliedFilters updates with a delay?

Yes. URL search parmas are not updated immediately. `setSearchParams` causes remix navigation, loader fetches new data and only after that search params are updated.

Reply:
Let's add something like:

```ts
// Allows updating the UI optimistically while Remix navigates to the URL
// with updated parameters.
const [filters, setFilters] = useState(appliedFilters);
```

It's possible that after changing filters in the UI and older version of `appliedFilters` will arrive and temporarily undo the changes, but I don't have any suggestions how to prevent that without keeping track of versions.